src/sds.c

```diff
@@ -1039,53 +1039,68 @@ int hex_digit_to_int(char c) {
     }
 }
 
+static inline int is_ptr_exceeds_end(const char *ptr, size_t len, const char *end) {
```

Comment:
Do we really need to distinguish the NULL case from the "not-exceed" case? I suggest we:
1. make both cases return the same value
2. make this function return bool type
3. document the behavior 

This way cases like:
`is_ptr_exceeds_end(p, 4, end) <= 0)` can be done as `!is_ptr_exceeds_end(p, 4, end)`
and cases like:
`is_ptr_exceeds_end(p, 0, end) > 0)` can be done as `is_ptr_exceeds_end(p, 0, end)`